src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,129 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidInputDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "my_valid_login";
+    private static final String VALID_PASSWORD = "valid#$PASS@!1234";
+    private static final int MINIMUM_LENGTH = 6;
+    private static final int VALID_AGE_MINIMUM = 18;
+    private static final int AGE_ZERO_INVALID = 0;
+
+    private static RegistrationServiceImpl registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+
+        user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE_MINIMUM);
+    }
+
+    @Test
+    void register_UserRegistered_Ok() {
+        assertEquals(user, registrationService.register(user));
+        assertTrue(Storage.people.contains(user));
+    }
+
+    @Test
+    void register_UserIsNull_notOk() {
+        String expected = "User can't be null!";
+        assertEquals(expected, assertException(null).getMessage());

```

Comment:
Usually, we do not do asserts inside asserts, it is hard to read and debug, but I guess this time you can leave it as it is